ion-hash/src/representation.rs

```diff
@@ -192,7 +194,9 @@ where
 {
     match value {
         None => {}
-        Some(timestamp) => hasher.encode_timestamp(timestamp)?,
+        Some(timestamp) => {
+            let _ = hasher.encode_timestamp(timestamp)?;
+        }
     };
```

Comment:
So minor as to almost be not worth mentioning, but this `let _ = ...` gives me a sad, would an `if let Some(timestamp) { ... }` avoid this problem or do you prefer the exhaustiveness of the match.

Reply:
I don't really need the `let _ = ...`. 

The reason for this change is the previous code had two arms that both returned unit, then this PR changed `encode_timestamp` to return usize. So, the problem is that the arm needed to return unit again. A `;` is all that's needed, i.e.:

```
Some(timestamp) => { hasher.encode_timestamp(timestamp)?; }
```

I *think* that's orthogonal to whether or not we choose to explicitly discard the usize result.